cdc server panic with new owner

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   run cdc replication with sysbench oltp_write_only data flow

2. What did you expect to see?

   cdc server runs normally

3. What did you see instead?


   cdc server panic


```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x16331aa]

goroutine 356 [running]:
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).CheckChangefeedNormal.func1(0x0, 0x203002, 0x203002, 0x203002, 0x0)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:265 +0x2a
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).PatchInfo.func1(0x0, 0x0, 0x413e05, 0xc000130540, 0xc0007235c0, 0x36, 0x1)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:292 +0xa2
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).patchAny.func1(0x0, 0x0, 0x0, 0x25, 0x4ca1560, 0x6b, 0x0, 0x0, 0x0)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:385 +0x13a
github.com/pingcap/ticdc/pkg/orchestrator.(*SingleDataPatch).Patch(0xc00a83be30, 0xc0084ba390, 0xc0084ba3f0, 0x0, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/interfaces.go:54 +0x82
github.com/pingcap/ticdc/pkg/orchestrator.(*EtcdWorker).applyPatches(0xc000710ee0, 0x7f022c4f3198, 0xc0006e6180, 0xc002e38870, 0x4, 0x9, 0xc0006da450, 0x10, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/etcd_worker.go:239 +0xcc
github.com/pingcap/ticdc/pkg/orchestrator.(*EtcdWorker).Run(0xc000710ee0, 0x7f022c4f3198, 0xc0006e6180, 0xc0006da450, 0x5f5e100, 0x0, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/etcd_worker.go:146 +0x87b
github.com/pingcap/ticdc/cdc/capture.(*Capture).runEtcdWorker(0xc000638680, 0x32484b8, 0xc0006e6180, 0x31db540, 0xc0004a4018, 0x32041f8, 0xc00093c1e0, 0x5f5e100, 0x0, 0x0)
	github.com/pingcap/ticdc/cdc/capture/capture.go:261 +0x135
github.com/pingcap/ticdc/cdc/capture.(*Capture).run.func3(0xc00094c090, 0xc000638680, 0x32484b8, 0xc0006e6180, 0xc000474310)
	github.com/pingcap/ticdc/cdc/capture/capture.go:180 +0x1ab
created by github.com/pingcap/ticdc/cdc/capture.(*Capture).run
	github.com/pingcap/ticdc/cdc/capture/capture.go:172 +0x2ed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x16331aa]

goroutine 294 [running]:
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).CheckChangefeedNormal.func1(0x0, 0x203002, 0x203002, 0x203002, 0x2)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:265 +0x2a
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).PatchInfo.func1(0x0, 0x0, 0x413e05, 0xc000053cc0, 0xc0087f2440, 0x36, 0x1)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:292 +0xa2
github.com/pingcap/ticdc/cdc/model.(*ChangefeedReactorState).patchAny.func1(0x0, 0x0, 0x0, 0x25, 0x4ca1560, 0x6b, 0x0, 0x0, 0x0)
	github.com/pingcap/ticdc/cdc/model/reactor_state.go:385 +0x13a
github.com/pingcap/ticdc/pkg/orchestrator.(*SingleDataPatch).Patch(0xc007d11788, 0xc00bf23ef0, 0xc00bf23f20, 0x0, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/interfaces.go:54 +0x82
github.com/pingcap/ticdc/pkg/orchestrator.(*EtcdWorker).applyPatches(0xc0007447e0, 0x7f6384146058, 0xc000718080, 0xc000739000, 0x4, 0x8, 0xc00050d020, 0x1, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/etcd_worker.go:239 +0xcc
github.com/pingcap/ticdc/pkg/orchestrator.(*EtcdWorker).Run(0xc0007447e0, 0x7f6384146058, 0xc000718080, 0xc00050d020, 0x5f5e100, 0x0, 0x0)
	github.com/pingcap/ticdc/pkg/orchestrator/etcd_worker.go:146 +0x87b
github.com/pingcap/ticdc/cdc/capture.(*Capture).runEtcdWorker(0xc000259f80, 0x32484b8, 0xc000718080, 0x31db540, 0xc000288258, 0x32041f8, 0xc000709080, 0x5f5e100, 0x0, 0x0)
	github.com/pingcap/ticdc/cdc/capture/capture.go:261 +0x135
github.com/pingcap/ticdc/cdc/capture.(*Capture).run.func3(0xc0000fec60, 0xc000259f80, 0x32484b8, 0xc000718080, 0xc000712070)
	github.com/pingcap/ticdc/cdc/capture/capture.go:180 +0x1ab
created by github.com/pingcap/ticdc/cdc/capture.(*Capture).run
	github.com/pingcap/ticdc/cdc/capture/capture.go:172 +0x2ed
```

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        v5.0.1

    - TiCDC version (execute `cdc version`):

        master@https://github.com/pingcap/ticdc/commit/f4388909e4f307fe1d0ed536de0a915d4014cab1

--- a/cdc/model/reactor_state.go
+++ b/cdc/model/reactor_state.go
@@ -262,7 +262,7 @@ func (s *ChangefeedReactorState) GetPatches() []orchestrator.DataPatch {
 func (s *ChangefeedReactorState) CheckChangefeedNormal() {
 	s.skipPatchesInThisTick = false
 	s.PatchInfo(func(info *ChangeFeedInfo) (*ChangeFeedInfo, bool, error) {
-		if info.AdminJobType.IsStopState() {
+		if info == nil || info.AdminJobType.IsStopState() {
 			s.skipPatchesInThisTick = true
 			return info, false, cerrors.ErrEtcdTryAgain.GenWithStackByArgs()
 		}

--- a/cdc/model/reactor_state_test.go
+++ b/cdc/model/reactor_state_test.go
@@ -652,6 +652,8 @@ func (s *stateSuite) TestCheckChangefeedNormal(c *check.C) {
 	defer testleak.AfterTest(c)()
 	state := NewChangefeedReactorState("test1")
 	stateTester := orchestrator.NewReactorStateTester(c, state, nil)
+	state.CheckChangefeedNormal()
+	stateTester.MustApplyPatches()
 	state.PatchInfo(func(info *ChangeFeedInfo) (*ChangeFeedInfo, bool, error) {
 		return &ChangeFeedInfo{SinkURI: "123", AdminJobType: AdminNone, Config: &config.ReplicaConfig{}}, true, nil
 	})

